src/core/net/netif.hpp

```diff
@@ -313,6 +322,7 @@ class Netif : public InstanceLocator, public LinkedListEntry<Netif>, private Non
      *
      */
     const NetifMulticastAddress *GetMulticastAddresses(void) const { return mMulticastAddresses.GetHead(); }
+    NetifMulticastAddress *      GetMulticastAddresses(void) { return mMulticastAddresses.GetHead(); }
```

Comment:
(Same as above) we do have some multicast entries in the multicast list which are literally `const` (stored in flash), again I know that the current use is checking for this and is ok, but providing a method that allows user to get non-const `NetifMulticastAddress` seems unsafe (allows possible incorrect use in future).

Reply:
removed